src/parser.cpp

```diff
@@ -346,7 +340,10 @@ skip_trailer_headers(
 
 template<class UInt>
 std::size_t
-clamp(UInt x, std::size_t limit) noexcept
+clamp(
+    UInt x,
+    std::size_t limit = std::numeric_limits<
+        std::size_t>::max()) noexcept
```

Comment:
You need to protect `max` from macros by putting parenthesis around the whole part before the parameter list